test/CSStrikes.t.sol

```diff
@@ -0,0 +1,279 @@
+import { IAssetRecovererLib } from "../src/lib/AssetRecovererLib.sol";
+import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+import { Fixtures } from "./helpers/Fixtures.sol";
+import { MerkleTree } from "./helpers/MerkleTree.sol";
+import { Stub } from "./helpers/mocks/Stub.sol";
+import { ERC20Testable } from "./helpers/ERCTestable.sol";
+import { Utilities } from "./helpers/Utilities.sol";
+import { InvariantAsserts } from "./helpers/InvariantAsserts.sol";
+
+using stdStorage for StdStorage;
+
+contract CSStrikesTestBase is Test, Fixtures, Utilities, InvariantAsserts {
+    address internal stranger;
+    address internal oracle;
+    address internal module;
+    CSStrikes internal strikes;
+    MerkleTree internal tree;
+
+    modifier assertInvariants() {
+        _;
+        vm.pauseGasMetering();
+        assertStrikesTree(strikes);
+        vm.resumeGasMetering();
+    }
+}
+
+contract CSStrikesConstructorTest is CSStrikesTestBase {
+    function setUp() public {
+        stranger = nextAddress("STRANGER");
+        oracle = nextAddress("ORACLE");
+        module = nextAddress("MODULE");
+    }
+
+    function test_constructor_happyPath() public {
+        strikes = new CSStrikes(module, oracle);
+        assertEq(strikes.ORACLE(), oracle);
+        assertEq(address(strikes.MODULE()), module);
+    }
+
+    function test_initialize_RevertWhen_ZeroModuleAddress() public {
+        vm.expectRevert(ICSStrikes.ZeroModuleAddress.selector);
+        new CSStrikes(address(0), oracle);
+    }
+
+    function test_initialize_RevertWhen_ZeroOracleAddress() public {
+        vm.expectRevert(ICSStrikes.ZeroOracleAddress.selector);
+        new CSStrikes(module, address(0));
+    }
+}
+
+contract CSStrikesTest is CSStrikesTestBase {
+    function setUp() public {
+        stranger = nextAddress("STRANGER");
+        oracle = nextAddress("ORACLE");
+        module = nextAddress("MODULE");
+
+        strikes = new CSStrikes(module, oracle);
+
+        tree = new MerkleTree();
+
+        vm.label(address(strikes), "STRIKES");
+    }
+
+    function test_hashLeaf() public assertInvariants {
+        uint256 noId = 42;
+        (bytes memory pubkey, ) = keysSignatures(1);
+        uint256[] memory strikesData = new uint256[](6);
+        strikesData[0] = 100500;
+
+        assertEq(
+            strikes.hashLeaf(noId, pubkey, strikesData),
+            0x01eba5ed7fb9c5ebb4262844c7125628afcbea15e57bc7f4dd0c80d34d633584
+        );
+    }
+
+    function test_verifyProof() public {
+        uint256 noId = 42;
+        (bytes memory pubkey, ) = keysSignatures(1);
+        uint256[] memory strikesData = new uint256[](6);
+        strikesData[0] = 100500;
+
+        tree.pushLeaf(abi.encode(noId, pubkey, strikesData));
+        tree.pushLeaf(abi.encode(noId + 1, pubkey, strikesData));
+
+        stdstore.target(address(strikes)).sig("treeRoot()").checked_write(
+            tree.root()
+        );
```

Comment:
why not just set it using `processOracleReport`? it doesn't look complicated

Reply:
just copy-pasted thing